### PR TITLE
Remote datasets: Add "load_earth_dist" to load "GSHHG Earth distance to shoreline" dataset

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

Add `load_earth_dist` to download the "GSHHG Earth distance to shoreline" dataset and load into a `xarray.DataArray`:

- Official documentation: https://www.generic-mapping-tools.org/remote-datasets/earth-dist.html
- Recipe: https://github.com/GenericMappingTools/gmtserver-admin/blob/master/recipes/earth_dist.recipe

**Adresses** #2431

**Preview**: https://pygmt-dev--3706.org.readthedocs.build/en/3706/api/generated/pygmt.datasets.load_earth_dist.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

**Comment by seisman** [comment]
Thanks for working on this function. I feel we should first update the upstream documentation to clarify the following points:

1. The distance unit is `km`
2. The distances have both negative and positive values. Positive means land to coastline and negative means ocean to coastline.
3. For the dataset image, maybe we should add the coastline to make the image more readable. Like this one:
map

File: pygmt/helpers/caching.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ def cache_data():
         # List of GMT remote datasets.
         "@earth_age_01d_g",
         "@earth_day_01d",
+        "@earth_dist_01d",
```
Comment (seisman) [inline pygmt/helpers/caching.py:17]: Also need to add `@N00W030.earth_dist_01m_g.nc` near line 48.

File: pygmt/tests/test_datasets_earth_dist.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    """
+    Test some properties of the Earth distance to shoreline 01d data.
+    """
+    data = load_earth_dist(resolution="01d")
+    assert data.name == "z"
+    assert data.attrs["description"] == "GSHHG Earth distance to shoreline"
+    assert data.attrs["units"] == "km"
+    assert data.attrs["horizontal_datum"] == "WGS84"
+    assert data.shape == (181, 361)
+    assert data.gmt.registration == 0
```
Comment (seisman) [inline pygmt/tests/test_datasets_earth_dist.py:20]: Need to update to the new syntax in #3696, depending on which PR is merged first.

File: pygmt/tests/test_datasets_earth_dist.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    """
+    Test loading low-resolution Earth distance to shoreline with "region".
+    """
+    data = load_earth_dist(resolution="01d", region=[-10, 10, -5, 5])
+    assert data.shape == (11, 21)
+    assert data.gmt.registration == 0
+    npt.assert_allclose(data.lat, np.arange(-5, 6, 1))
+    npt.assert_allclose(data.lon, np.arange(-10, 11, 1))
+    npt.assert_allclose(data.min(), -1081.94, atol=0.01)
+    npt.assert_allclose(data.max(), 105.17999, atol=0.01)
```
Comment (seisman) [inline pygmt/tests/test_datasets_earth_dist.py:37]:
```suggestion
    npt.assert_allclose(data.max(), 105.18, atol=0.01)
```

File: pygmt/tests/test_datasets_earth_dist.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    """
+    data = load_earth_dist(resolution="01m", region=[-10, -9, 3, 5])
+    assert data.shape == (121, 61)
+    assert data.gmt.registration == 0
+    assert data.coords["lat"].data.min() == 3.0
+    assert data.coords["lat"].data.max() == 5.0
+    assert data.coords["lon"].data.min() == -10.0
+    assert data.coords["lon"].data.max() == -9.0
+    npt.assert_allclose(data.min(), -243.62, atol=0.01)
+    npt.assert_allclose(data.max(), 2.939995, atol=0.01)
```
Comment (seisman) [inline pygmt/tests/test_datasets_earth_dist.py:53]:
```suggestion
    npt.assert_allclose(data.max(), 2.94, atol=0.01)
```

File: pygmt/helpers/caching.py (hunk 1 above)
Comment (yvonnefroehlich) [reply pygmt/helpers/caching.py:17]: Add it in commit https://github.com/GenericMappingTools/pygmt/pull/3706/commits/2404d9c0f2bd78c2c189553a34956da044cc5377.

File: pygmt/datasets/earth_dist.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        higher than 5 arc-minutes (i.e., ``"05m"``).
+    registration
+        Grid registration type. Either ``"pixel"`` for pixel registration or
+        ``"gridline"`` for gridline registration.
+
+    Returns
+    -------
+    grid
+        The GSHHG Earth distance to shoreline grid. Coordinates are latitude and
+        longitude in degrees. Distances are in kilometers.
```
Comment (seisman) [inline pygmt/datasets/earth_dist.py:70]:
```suggestion
        The GSHHG Earth distance to shoreline grid. Coordinates are latitude and
        longitude in degrees. Distances are in kilometers. Positive (negative) 
        distance values mean land to coastline (ocean to coastline).
```

Comment (yvonnefroehlich) [reply pygmt/datasets/earth_dist.py]: See commit https://github.com/GenericMappingTools/pygmt/pull/3706/commits/0682a1042b0bb123d2d092dc5da5a7536576aa98 for a slightly modified formulation.

**Comment by yvonnefroehlich** [comment]
Currently we use both, the abbreviation and the complete word for units in `load_remote_datasets.py`:

https://github.com/GenericMappingTools/pygmt/blob/b90fa469a2fd6cd1ec25e4ed99499eb3c48589e8/pygmt/datasets/load_remote_dataset.py#L113-L115

but

https://github.com/GenericMappingTools/pygmt/blob/b90fa469a2fd6cd1ec25e4ed99499eb3c48589e8/pygmt/datasets/load_remote_dataset.py#L135-L137

I think we should decide on one way and update the dictionary in `load_remote_datasets.py` and the related tests.

**Comment by seisman** [comment]
That's a good point. The [CF convention](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.12/cf-conventions.html) mentions that the unit string must be recognized by the [UDUNIT](https://www.unidata.ucar.edu/software/udunits/) package. I quickly read the UNUNIT documentation, I think valid values are `meter`, `m` and `metre` (without `s`). But the CF-convention has an example that uses `meters` (https://cfconventions.org/Data/cf-conventions/cf-conventions-1.12/cf-conventions.html#vertical-coordinate).

**Comment by seisman** [comment]
This section (http://cfconventions.org/Data/cf-conventions/cf-conventions-1.12/cf-conventions.html#_dimensional_vertical_coordinate) says:

> Plural forms are also acceptable.

So I think `meters`/`meter`/`m` are fine. Let's use `meters` then.

**Comment by yvonnefroehlich** [comment]
> This section (http://cfconventions.org/Data/cf-conventions/cf-conventions-1.12/cf-conventions.html#_dimensional_vertical_coordinate) says:
> 
> > Plural forms are also acceptable.
> 
> So I think `meters`/`meter`/`m` are fine. Let's use `meters` then.

Submitted PR #3725 for updating this for all datasets.

**Pull request merged**